src/test/java/apimatic/core/logger/LoggerUtilitiesTest.java

```diff
@@ -0,0 +1,104 @@
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+
+import io.apimatic.core.logger.LoggerUtilities;
+
+public class LoggerUtilitiesTest {
+    @Test
+    public void testExtractHeadersToLogIncludeHeaders() {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Header1", "Value1");
+        headers.put("Header2", "Value2");
+        headers.put("Header3", "Value3");
+
+        List<String> headersToInclude = getLowerCaseList("Header1", "Header2");
+
+        Map<String, String> extractedHeaders = LoggerUtilities.extractHeadersToLog(headers,
+                headersToInclude, Collections.emptyList());
+
+        final int expectedHeaderSize = 2;
+        assertEquals(expectedHeaderSize, extractedHeaders.size());
+        assertTrue(extractedHeaders.containsKey("Header1"));
+        assertTrue(extractedHeaders.containsKey("Header2"));
+    }
+
+    @Test
+    public void testExtractHeadersToLogExcludeHeaders() {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Header1", "Value1");
+        headers.put("Header2", "Value2");
+        headers.put("Header3", "Value3");
+
+        List<String> headersToExclude = getLowerCaseList("Header2", "Header3");
+
+        Map<String, String> extractedHeaders = LoggerUtilities.extractHeadersToLog(headers,
+                Collections.emptyList(), headersToExclude);
+
+        final int expectedHeaderSize = 1;
+        assertEquals(expectedHeaderSize, extractedHeaders.size());
+        assertTrue(extractedHeaders.containsKey("Header1"));
+    }
+
+    @Test
+    public void testExtractHeadersToLogNoFilter() {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Header1", "Value1");
+        headers.put("Header2", "Value2");
+        headers.put("Header3", "Value3");
+
+        Map<String, String> extractedHeaders = LoggerUtilities.extractHeadersToLog(headers,
+                Collections.emptyList(), Collections.emptyList());
+
+        final int expectedHeaderSize = 3;
+        assertEquals(expectedHeaderSize, extractedHeaders.size());
+        assertTrue(extractedHeaders.containsKey("Header1"));
+        assertTrue(extractedHeaders.containsKey("Header2"));
+        assertTrue(extractedHeaders.containsKey("Header3"));
+    }
+
+    @Test
+    public void testExtractHeadersToLogWithFilter() {
+        final boolean enableMasking = true;
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Accept", "Value1");
+        headers.put("Masked-Header", "MaskedValue");
+
+        List<String> headersToWhiteList = getLowerCaseList("Accept");
+
+        Map<String, String> extractedHeaders = LoggerUtilities.filterSensitiveHeaders(headers,
+                headersToWhiteList, enableMasking);
+
+        final int expectedHeaderSize = 2;
+        final String expectedAcceptHeaderValue = "Value1";
+        final String expectedPoweredByHeaderValue = "**Redacted**";
```

Comment:
`expectedPoweredByHeaderValue` should be renamed to `expectedMaskedHeaderValue`

Reply:
variable has been renamed